staging/src/k8s.io/apiserver/pkg/cel/types.go

```diff
@@ -357,9 +357,19 @@ type RuleTypes struct {
 //
 // If the RuleTypes value is nil, an empty []cel.EnvOption set is returned.
 func (rt *RuleTypes) EnvOptions(tp ref.TypeProvider) ([]cel.EnvOption, error) {
-	if rt == nil {
-		return []cel.EnvOption{}, nil
+	rtWithTypes, err := rt.WithTypeProvider(tp)
+	if err != nil {
+		return nil, err
 	}
+	return []cel.EnvOption{
+		cel.CustomTypeProvider(rtWithTypes),
+		cel.CustomTypeAdapter(rtWithTypes),
+		cel.Variable("rule", rt.ruleSchemaDeclTypes.root.CelType()),
+	}, nil
+}
+
+// WithTypeProvider returns a new RuleTypes that sets the given TypeProvider
+func (rt *RuleTypes) WithTypeProvider(tp ref.TypeProvider) (*RuleTypes, error) {
```

Comment:
Do we need to check rt nil here?

Reply:
There is no existing use case for a nil rt, but checked just in case.